Tests/UnitTests/Core/Propagator/CovarianceTransportTests.cpp

```diff
@@ -0,0 +1,108 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <boost/test/unit_test.hpp>
+
+#include "Acts/Propagator/CovarianceTransport.hpp"
+#include "Acts/Surfaces/PlaneSurface.hpp"
+
+namespace tt = boost::test_tools;
+
+namespace Acts {
+namespace Test {
+
+BOOST_AUTO_TEST_CASE(covariance_cache_invalid) {
+  CovarianceCache covCache;
+  BOOST_CHECK(covCache.applyTransport == false);
+}
+
+BOOST_AUTO_TEST_CASE(covariance_cache_bound) {
+  // Create a test context
+  GeometryContext tgContext = GeometryContext();
+
+  Vector3 position{1., 2., 3.};
+  double time = 4.;
+  Vector3 direction = Vector3(5., 6., 7.).normalized();
+  auto planeSurface = Surface::makeShared<PlaneSurface>(position, direction);
+
+  double qop = 0.125;
+
+  BoundVector boundParameters;
+  boundParameters << 0., 0., VectorHelpers::phi(direction),
+      VectorHelpers::theta(direction), qop, time;
+
+  BoundSymMatrix boundCovariance = 8. * BoundSymMatrix::Identity();
+
+  CovarianceCache covCache(tgContext, *planeSurface, position, boundParameters,
+                           boundCovariance);
+  // Test that the transport should be applied now
+  BOOST_CHECK(covCache.applyTransport == true);
+  // Test that the set covariance is 5x5 and what it has been set
+  BOOST_CHECK_EQUAL(std::get<BoundSymMatrix>(covCache.covariance),
+                    boundCovariance);
+  BOOST_CHECK_THROW(std::get<FreeSymMatrix>(covCache.covariance),
+                    std::bad_variant_access);
+  // Test that the bound to free jacobian has a value
+  BOOST_CHECK(covCache.boundToFreeJacobian.has_value());
+  BOOST_CHECK(not covCache.boundToFreeJacobian.value().isApprox(
+      BoundToFreeMatrix::Zero()));
+  // Test that the free transport jacobian, derivative is properly set up
+  BOOST_CHECK_EQUAL(covCache.freeTransportJacobian, FreeMatrix::Identity());
+  BOOST_CHECK_EQUAL(covCache.freePathDerivatives, FreeVector::Zero());
```

Comment:
Although this is correct within this implementation, it could be any value in general. I'm not sure whether this test is reasonable.

Reply:
See above, this file will still change substantially.